python/tskit/trees.py

```diff
@@ -3645,6 +3646,53 @@ def write_vcf(
         )
         writer.write(output)
 
+    def write_ms(
+        self,
+        output,
+        print_trees=False,
+        precision=3,
+        mutation_rate=0,
+        num_replicates=0
+    ):
+        """
+        Write ``ms`` formatted output from a tree sequence. Since this is a method of 
+        the TreeSequence class, there are two possible calls to this function 
+        depending on whether the ``num_replicates`` argument is used in msprime.simulate 
+        If the num_replicates argument is not used, a single tree sequence is generated.
+        The write_ms function is then called as:
+        .. code-block:: python
+            
+            >>> tree_sequence = msprime.simulate(<simulation arguments>)
+            >>> with open('output.ms', 'w') as ms_file:
+            >>>     tree_sequence.write_ms(ms_file, mutation_rate=mutation_rate)
+
+        If the num_replicates argument is used in msprime.simulate, a list of tree 
+        sequences is generated. The call to write_ms then takes the form:
+        .. code-block:: python
+
+            >>> tree_sequences = msprime.simulate(<simulation arguments>, num_replicates=num_replicates)
+            >>> with open('output.ms', 'w') as ms_file:
+            >>> for tree_sequence in tree_sequences:
+            >>>     tree_sequence.write_ms(ms_file, mutation_rate=mutation_rate, num_replicates=num_replicates)
```

Comment:
You've got a nice solution to this. But really, `write_ms` is not a property of a TreeSequence, it's a thing that you can do to one, or maybe many, tree sequences. What if instead of `ts.write_ms` you did `tskit.write_ms(ts)`? That way if ts is a TreeSequence, you'd write it out, and if `ts` is a generator (of tree sequences) then you have replicates, and write them out?